pkg/backend/metrics.go

```diff
@@ -0,0 +1,203 @@
+	} else {
+		dimensions["fips"] = unknown
+	}
+}
+
+func (ocb *openShiftClusterBackend) gatherAuthMetrics(doc *api.OpenShiftClusterDocument, dimensions map[string]string) {
+	if doc.OpenShiftCluster.Properties.PlatformWorkloadIdentityProfile != nil {
+		dimensions["workloadidentity"] = enabled
+	} else if doc.OpenShiftCluster.Properties.ServicePrincipalProfile != nil {
+		dimensions["serviceprincipal"] = disabled
+	} else {
+		dimensions["serviceprincipal"] = unknown
+	}
+
+	if doc.OpenShiftCluster.Properties.ClusterProfile.PullSecret != "" {
+		dimensions["pullsecret"] = enabled
+	} else {
+		dimensions["pullsecret"] = disabled
+	}
+}
+
+func (ocb *openShiftClusterBackend) gatherNetworkMetrics(doc *api.OpenShiftClusterDocument, dimensions map[string]string) {
+	for _, p := range doc.OpenShiftCluster.Properties.IngressProfiles {
+		if p.Visibility == api.VisibilityPrivate {
+			dimensions["ingressprofile"] = fmt.Sprintf("%s.%s", string(api.VisibilityPrivate), p.Name)
+		} else if p.Visibility == api.VisibilityPublic {
+			dimensions["ingressprofile"] = fmt.Sprintf("%s.%s", string(api.VisibilityPublic), p.Name)
+		} else {
+			dimensions["ingressprofile"] = unknown
+		}
+	}
+
+	if doc.OpenShiftCluster.Properties.NetworkProfile.OutboundType == api.OutboundTypeUserDefinedRouting {
+		dimensions["networkprofile.outboundtype"] = string(api.OutboundTypeUserDefinedRouting)
+	} else if doc.OpenShiftCluster.Properties.NetworkProfile.OutboundType == api.OutboundTypeLoadbalancer {
+		dimensions["networkprofile.outboundtype"] = string(api.OutboundTypeLoadbalancer)
+	} else {
+		dimensions["networkprofile.outboundtype"] = unknown
+	}
+
+	if doc.OpenShiftCluster.Properties.NetworkProfile.PodCIDR != "10.128.0.0/14" {
+		dimensions["podcidr"] = custom
+	} else {
+		dimensions["podcidr"] = defaultSet
+	}
+
+	if doc.OpenShiftCluster.Properties.NetworkProfile.ServiceCIDR != "172.30.0.0/16" {
+		dimensions["servicecidr"] = custom
+	} else {
+		dimensions["servicecidr"] = defaultSet
+	}
+
+	if doc.OpenShiftCluster.Properties.ClusterProfile.Domain != "" {
+		dimensions["domain"] = enabled
+	}
+
+	if doc.OpenShiftCluster.Properties.NetworkProfile.LoadBalancerProfile.ManagedOutboundIPs != nil {
+		dimensions["managedoutboundips"] = strconv.FormatInt(int64(doc.OpenShiftCluster.Properties.NetworkProfile.LoadBalancerProfile.ManagedOutboundIPs.Count), 10)
+	}
+
+	if doc.OpenShiftCluster.Properties.NetworkProfile.PreconfiguredNSG == api.PreconfiguredNSGEnabled {
+		dimensions["preconfigurednsg"] = string(api.PreconfiguredNSGEnabled)
+	} else if doc.OpenShiftCluster.Properties.NetworkProfile.PreconfiguredNSG == api.PreconfiguredNSGDisabled {
+		dimensions["preconfigurednsg"] = string(api.PreconfiguredNSGDisabled)
+	} else {
+		dimensions["preconfigurednsg"] = unknown
+	}
+
+	if doc.OpenShiftCluster.Properties.FeatureProfile.GatewayEnabled {
+		dimensions["gatewayenabled"] = enabled
+	} else {
+		dimensions["gatewayenabled"] = disabled
+	}
```

Comment:
Do we need this? The default state for all new cluster installs is enabled.

Reply:
It makes sense to me to include it, since these metrics are being emitted on all operations and not just on new cluster creations.